versions/models.py

```diff
@@ -88,17 +100,15 @@ def next_version(self, object):
         if object.version_end_date == None:
             return object
         else:
-            try:
-                next = self.get(
-                    Q(identity=object.identity),
-                    Q(version_start_date=object.version_end_date))
-            except MultipleObjectsReturned as e:
-                raise MultipleObjectsReturned(
-                    "next_version couldn't uniquely identify the next version of object " + str(
-                        object.identity) + " to be returned\n" + str(e))
-            except ObjectDoesNotExist as e:
+            next = self.filter(
+                Q(identity=object.identity),
+                Q(version_start_date__gte=object.version_end_date)
+            ).order_by('version_start_date').first()
```

Comment:
I see your point that conflicts only appear when navigating through versions. It's a valid one and may be very confusing when it appears.
Therefore, a DB-level constraint would be better. But I don't like to introduce additional dependencies.
For the moment, I think we can leave code as is. 
What do you suggest on how to proceed with this subject?